youtube_dl/jsinterp.py

```diff
@@ -18,13 +22,75 @@
     ('+', operator.add),
     ('%', operator.mod),
     ('/', operator.truediv),
-    ('*', operator.mul),
+    ('*', operator.mul)
 ]
 _ASSIGN_OPERATORS = [(op + '=', opfunc) for op, opfunc in _OPERATORS]
 _ASSIGN_OPERATORS.append(('=', lambda cur, right: right))
 
+_RESERVED_RE = r'(?:function|var|(?P<ret>return))\s'
+
+_OPERATORS_RE = r'|'.join(re.escape(op) for op, opfunc in _OPERATORS)
+_ASSIGN_OPERATORS_RE = r'|'.join(re.escape(op) for op, opfunc in _ASSIGN_OPERATORS)
+
 _NAME_RE = r'[a-zA-Z_$][a-zA-Z_$0-9]*'
 
+_SINGLE_QUOTED = r"""'(?:[^'\\\\]*(?:\\\\\\\\|\\\\['"nurtbfx/\\n]))*[^'\\\\]*'"""
+_DOUBLE_QUOTED = r'''"(?:[^"\\\\]*(?:\\\\\\\\|\\\\['"nurtbfx/\\n]))*[^"\\\\]*"'''
+_STRING_RE = r'%s|%s' % (_SINGLE_QUOTED, _DOUBLE_QUOTED)
+
+_INTEGER_RE = r'%(hex)s|%(dec)s|%(oct)s' % {'hex': __HEXADECIMAL_RE, 'dec': __DECIMAL_RE, 'oct': __OCTAL_RE}
+_FLOAT_RE = r'%(dec)s\.%(dec)s' % {'dec': __DECIMAL_RE}
+
+_BOOL_RE = r'true|false'
+# XXX: it seams group cannot be refed this way
+# r'/(?=[^*])[^/\n]*/(?![gimy]*(?P<reflag>[gimy])[gimy]*\g<reflag>)[gimy]{0,4}'
+_REGEX_RE = r'/(?=[^*])[^/\n]*/[gimy]{0,4}'
```

Comment:
> They can't be multiline, can they?

Yep. According to [ECMA 262 5.1](http://www.ecma-international.org/ecma-262/5.1/), CR (U+000D), LF (U+000A), LS (U+2028) and PS (U+2029) are not allowed in RegExp literals